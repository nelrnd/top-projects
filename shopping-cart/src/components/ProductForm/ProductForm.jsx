import { useState } from "react"
import styles from "./ProductForm.module.css"
import { useOutletContext } from "react-router-dom"
import QuantityController from "../QuantityController/QuantityController"

export default function ProductForm({ product }) {
  const { addToCart } = useOutletContext()

  const [quantity, setQuantity] = useState(1)

  const handleSubmit = (event) => {
    event.preventDefault()
    addToCart(product.id, quantity)
  }

  return (
    <form onSubmit={handleSubmit} className={styles.ProductForm}>
      <QuantityController
        quantity={quantity}
        onIncr={() => setQuantity((prev) => Number(prev) + 1)}
        onDecr={() => setQuantity((prev) => Number(prev) - 1)}
        onChange={(event) => setQuantity(event.target.value)}
      />
      <button className="btn primary-btn">Add to cart</button>
    </form>
  )
}
