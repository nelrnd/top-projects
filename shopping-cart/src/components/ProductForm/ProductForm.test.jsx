import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import ProductForm from "./ProductForm"
import { vi } from "vitest"

const product = {
  id: 1,
  title: "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops",
  price: 109.95,
  description:
    "Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday",
  category: "men's clothing",
  image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
  rating: {
    rate: 3.9,
    count: 120,
  },
}

describe("ProductForm component", () => {
  it("renders ProductForm", () => {
    const { container } = render(<ProductForm product={product} />)
    expect(container).toMatchSnapshot()
  })

  /*

  it("adds one item when clicking button", async () => {
    vi.mock("../../CartProvider", () => ({ addToCart: () => {} }))
    const user = userEvent.setup()
    render(<ProductForm product={product} />)
    const addButton = screen.getByRole("button", { name: "Add to cart" })
    expect(addButton).toBeInTheDocument()
    await user.click(addButton)
  })

  */
})
