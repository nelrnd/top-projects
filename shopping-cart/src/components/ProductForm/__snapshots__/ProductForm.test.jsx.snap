// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductForm component > renders ProductForm 1`] = `
<div>
  <form
    class="_ProductForm_fb947a"
  >
    <div
      class="_QuantityController_b7ada0"
    >
      <button
        type="button"
      >
        -
      </button>
      <input
        type="number"
        value="1"
      />
      <button
        type="button"
      >
        +
      </button>
    </div>
    <button
      class="btn primary-btn"
    >
      Add to cart
    </button>
  </form>
</div>
`;
